ipld/blockstore/src/lib.rs

```diff
@@ -1,40 +1,49 @@
 use std::rc::Rc;
 
 use anyhow::Result;
-use cid::{multihash, Cid};
+use cid::multihash::MultihashDigest;
+use cid::CidGeneric;
 
 pub mod tracking;
 
 mod memory;
 pub use memory::MemoryBlockstore;
 
 mod block;
-pub use block::*;
+pub use block::Block;
+
+/// The default size for multihashes is currently set to `64` as it's the value the `multihash` is
+/// using is its default code table. Once a custom code table is used, it could be reduced to `32`.
+const DEFAULT_MULTIHASH_ALLOC_SIZE: usize = 64;
 
 /// An IPLD blockstore suitable for injection into the FVM.
 ///
 /// The cgo blockstore adapter implements this trait.
-pub trait Blockstore {
+///
+pub trait Blockstore<const S: usize = DEFAULT_MULTIHASH_ALLOC_SIZE> {
```

Comment:
Being generic over the code type makes sense, but being generic over the size really isn't worth it:

1. It will lead to an annoying refactor in the actors (which will have to be backported).
2. It will conflict with all open PRs.

Reply:
Agreed, it didn't really felt nice anyway. I'll look if I find a better, less intrusive way.